File: packages/runtime/nodejs/src/workflow/type/core/engine.ts
import { ITask } from './task';
import { IExecutor } from './executor';
import { InvokeParams } from '../base';

export interface EngineServices {
  Executor: IExecutor;
}

export interface IEngine {
  invoke(params: InvokeParams): ITask;
}

export const IEngine = Symbol.for('Engine');

File: packages/runtime-go/interface/runtime/engine.go
/**
 * Copyright (c) 2025 Bytedance Ltd. and/or its affiliates
 * SPDX-License-Identifier: MIT
 */

package runtimeType

type EngineServices struct {
	Validation IValidation
	Executor   IExecutor
}

type IEngine interface {
	Invoke(params InvokeParams) ITask
	ExecuteNode(context IContext, node INode[any]) error
}

const EngineKey = "Engine"
